src/views/settings/Auth/OIDC.vue

```diff
@@ -135,6 +129,61 @@ export default {
         }
       }
     }
+  },
+  methods: {
+    async onProviderEndpointChange(value, updateForm) {
+      let data = {}
+      try {
+        data = await this.discovery(value)
+      } catch (err) {
+        data = this.setDefault(value)
+      }
+      updateForm(data)
+    },
+    async discovery(issuer, updateForm) {
+      if (!issuer.startsWith('http')) {
+        throw new Error('Invalid issuer')
+      }
+      const url = `${issuer}/.well-known/openid-configuration`
+      const config = {
+        ignoreSSLVerification: true,
+        disableFlashErrorMsg: true
+      }
+      const configMap = {
+        AUTH_OPENID_PROVIDER_AUTHORIZATION_ENDPOINT: 'authorization_endpoint',
+        AUTH_OPENID_PROVIDER_TOKEN_ENDPOINT: 'token_endpoint',
+        AUTH_OPENID_PROVIDER_JWKS_ENDPOINT: 'jwks_uri',
+        AUTH_OPENID_PROVIDER_USERINFO_ENDPOINT: 'userinfo_endpoint',
+        AUTH_OPENID_PROVIDER_END_SESSION_ENDPOINT: 'end_session_endpoint'
+      }
+      try {
+        const res = await this.$axios.get(url, config)
+        const data = {}
+        for (const [k, v] of Object.entries(configMap)) {
+          data[k] = res[v]
+        }
+        if (Object.keys(data).length < 5) {
+          throw new Error('Invalid issuer, missing required fields')
+        }
+        return data
+      } catch (err) {
+        throw err
+      }
+    },
+    setDefault(issuer) {
+      const fields = {
+        AUTH_OPENID_PROVIDER_AUTHORIZATION_ENDPOINT: 'authorize',
+        AUTH_OPENID_PROVIDER_TOKEN_ENDPOINT: 'token',
+        AUTH_OPENID_PROVIDER_JWKS_ENDPOINT: 'jwks',
+        AUTH_OPENID_PROVIDER_USERINFO_ENDPOINT: 'userinfo',
+        AUTH_OPENID_PROVIDER_END_SESSION_ENDPOINT: 'logout'
+      }
+      const data = {}
+      for (const [k, v] of Object.entries(fields)) {
+        data[k] = issuer + '/' + v
+      }
+      return data
+    }
   }
 }
 </script>
```

Comment:
I see no obvious issues with this code snippet, and everything looks correct based on the knowledge cutoff provided. However, without knowing more context about how it's intended to be used, any further analysis might not apply accurately. The structure appears well organized in terms of importing necessary components and data properties.

Should there be specific concerns or areas that you are interested in discussing, such as performance enhancements, security checks or optimizations related to user experience? Feel free to share those details so I can assist you better!